core/cfgwatcher.go

```diff
@@ -0,0 +1,53 @@
+package core
+
+import (
+	"log"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type configWatchdog struct {
+	path    string
+	watcher *fsnotify.Watcher
+	notify  chan struct{}
+}
+
+// NewConfigWatchdog creates new file watcher for config
+// directory
+func NewConfigWatchdog(path string) (*configWatchdog, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+	err = watcher.Add(path)
+
+	return &configWatchdog{path, watcher, make(chan struct{})}, err
+}
+
+// Start config directory watching
+func (cw *configWatchdog) Start() {
+	go func() {
```

Comment:
According to https://github.com/fsnotify/fsnotify#faq -> https://github.com/howeyc/fsnotify/issues/7 -> https://github.com/golang/go/issues/8282#c4 it might be unsafe to watch Events and Errors on the same chan.